app/models/article.rb

```diff
@@ -28,7 +28,16 @@ class Article < ApplicationRecord
   counter_culture :user
   counter_culture :organization
 
-  has_many :comments, as: :commentable, inverse_of: :commentable
+  has_many :buffer_updates, dependent: :destroy
+  has_many :comments, as: :commentable, inverse_of: :commentable, dependent: :nullify
```

Comment:
this is important. When articles are deleted we leave `commentable_id` and `commentable_type` with the previous value, but those 2 values are invalid. The reason why we haven't noticed until now is that our validation is counter intuitive.

See my comment in the `comment.rb` file for further info